capstone/backend/src/main/java/com/google/sps/servlets/CreateOrUpdateUserlistServlet.java

```diff
@@ -37,13 +33,23 @@ public ResponseBody(ResponseStatus responseStatus) {
  @Override
  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
    Gson g = new Gson();
-   RequestBody requestBody = getRequestBody(request);
-   DatabaseClient dbClient = initClient();
-   writeData(requestBody, dbClient);
-
-   ResponseBody responseBody = new ResponseBody(ResponseStatus.SUCCESS);
-   response.setContentType("application/json;");
-   response.getWriter().println(g.toJson(responseBody));
+   SpannerUtilFunctions suf = new SpannerUtilFunctions();
+   try {
+     RequestBody requestBody = getRequestBody(request);
+   } catch (IOException) {
+      response.setStatus(BAD_REQUEST)
+      response.getWriter().println("");
+      return;
+   }
+   try {
+     suf.writeUserLists(requestBody.userId. requestBody.UserList.listId, requestBody.UserList.itemTypes);
+   } catch (Exception e) {
```

Comment:
Try to figure out the more specific exception? also in your function declaration on line 34 you're only throwing IOException.

Reply:
I left that there because I didn't know what exception the database specific functions were going to throw. I changed it to check for the return value instead. This way the database wrapper functions can return false when there is an error.